framework/src/ble_manager/ble_manager_state.c

```diff
@@ -109,32 +110,55 @@ static ble_result_e _convert_ret(trble_result_e val)
 	return (ble_result_e)val;
 }
 
-static int _bledev_handler(int argc, char *argv[])
-{
-	g_run_process = 1;
-	ble_scanned_device *scanned_device;
-
-	while (g_run_process) {
-		if (sem_wait(&scan_queue->countsem) < 0) {
-			if (errno == EINTR) {
-				continue;
-			} else {
-				g_run_process = 0;
-				return -1;
+static void _event_caller(int evt_pri, void *data) {
+	switch (evt_pri) {
+	case BLE_QUEUE_EVT_PRI_HIGH: {
+		blemgr_msg_params *msg = (blemgr_msg_params *)data;
+		switch (msg->evt) {
+		case BLE_EVT_CLIENT_CONNECT: {
+			ble_client_device_connected_cb callback = msg->param[0];
+			callback(msg->param[1], msg->param[2]);
+		} break;
+		case BLE_EVT_CLIENT_DISCONNECT: {
+			ble_client_device_disconnected_cb callback = msg->param[0];
+			callback(msg->param[1]);
+		} break;
+		case BLE_EVT_CLIENT_NOTI: {
+			ble_client_operation_notification_cb callback = msg->param[0];
+			ble_attr_handle attr_handle = *(ble_attr_handle *)(msg->param[2] + sizeof(ble_conn_handle));
+
+			ble_data read_result[1];
+			read_result->length = *(uint16_t *)(msg->param[2] + sizeof(ble_conn_handle) + sizeof(ble_attr_handle));
+			read_result->data = (uint8_t *)(msg->param[2] + sizeof(ble_conn_handle) + sizeof(ble_attr_handle) + sizeof(read_result->length));
+			
+			callback(msg->param[1], attr_handle, read_result);
+		} break;
+		case BLE_EVT_SCAN_STATE: {
+			ble_scan_state_e state = *(ble_scan_state_e *)msg->param[2];
+			ble_client_scan_state_changed_cb callback = msg->param[0];
+			callback(state);
+			if (state == BLE_SCAN_STOPPED) {
+				memset(&(g_scan_ctx.filter), 0, sizeof(ble_scan_filter));
+				memset(&(g_scan_ctx.callback), 0, sizeof(ble_scan_callback_list));
 			}
+		} break;
+		default:
+			break;
 		}
-		if (scan_queue->read_index == scan_queue->write_index) {
-			BLE_LOG_INFO("[BLEMGR] Scan Info Queue is Empty\n");
-			continue;
+
+		if (msg->param[2] != NULL) {
+			free(msg->param[2]);
```

Comment:
I think it is better to set NULL after `free`.
Because `free` will only release the memory, but not set NULL.

Reply:
"msg->param[2]" is just temporary variable. That would never be used after "_event_caller" is done.
Hence, I think making that be NULL is not necessary.